api/src/test/java/org/openmrs/CohortTest.java

```diff
@@ -156,6 +156,89 @@ public void intersect_shouldContainVoidedAndExpiredMemberships() throws Exceptio
 		});
 	}
 
+	@Test
+	public void intersect_shouldContainTheEarliestMembershipWithSamePatientId() throws Exception {
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		Date endDate = dateFormat.parse("2017-02-01 00:00:00");
+
+		Cohort cohortOne = new Cohort(3);
+		CohortMembership membershipOne = new CohortMembership(7, startDate);
+		membershipOne.setEndDate(endDate);
+		cohortOne.addMembership(membershipOne);
+
+
+		Cohort cohortTwo = new Cohort(4);
+		CohortMembership membershipTwo = new CohortMembership(8, startDate);
+		membershipTwo.setEndDate(endDate);
+		cohortTwo.addMembership(membershipOne);
+		cohortTwo.addMembership(membershipTwo);
+		startDate = dateFormat.parse("2017-01-02 00:00:00");
+		membershipTwo = new CohortMembership(7, startDate);
```

Comment:
I am a bit curious as to why memberShipTwo is being reassigned over here

Reply:
I was using memberShipTwo to store CohortMemberships . I added the membership in the cohort's membership list and then we don't need another reference to this membership but we want to add more memberships in the list so i create a new object. I reassign memberShipTwo  because i see no point in creating new variable and i use this auxiliar to be more clear what is added.